stravalib/model.py

```diff
@@ -417,32 +247,74 @@ class AthleteStats(
     profile. Non-public activities are not counted for these totals.
     """
 
+    # field overrides from superclass for type extensions:
+    recent_ride_totals: Optional[ActivityTotals] = None
+    recent_run_totals: Optional[ActivityTotals] = None
+    recent_swim_totals: Optional[ActivityTotals] = None
+    ytd_ride_totals: Optional[ActivityTotals] = None
+    ytd_run_totals: Optional[ActivityTotals] = None
+    ytd_swim_totals: Optional[ActivityTotals] = None
+    all_ride_totals: Optional[ActivityTotals] = None
+    all_run_totals: Optional[ActivityTotals] = None
+    all_swim_totals: Optional[ActivityTotals] = None
+
     _field_conversions = {
         "biggest_ride_distance": uh.meters,
         "biggest_climb_elevation_gain": uh.meters,
     }
 
-    _activity_total_extensions = extend_types(
-        "recent_ride_totals",
-        "recent_run_totals",
-        "recent_swim_totals",
-        "ytd_ride_totals",
-        "ytd_run_totals",
-        "ytd_swim_totals",
-        "all_ride_totals",
-        "all_run_totals",
-        "all_swim_totals",
-        model_class=ActivityTotals
-    )
-
 
 class Athlete(
     DetailedAthlete, DeprecatedSerializableMixin, BackwardCompatibilityMixin, BoundClientEntity
 ):
-    is_authenticated: Optional[bool] = None
+    # field overrides from superclass for type extensions:
+    clubs: Optional[List[Club]] = None
+    bikes: Optional[List[Bike]] = None
+    shoes: Optional[List[Shoe]] = None
```

Comment:
So it turns out that one can simply override a superclass field to extend its type. This allows simplification of the model and removal of the `extend_types()` custom validator factory.